AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/Protocols/AmplifyAuthTask.swift

```diff
@@ -25,14 +25,17 @@ protocol AmplifyAuthTask {
 
 }
 
-extension AmplifyAuthTask {
+extension AmplifyAuthTask where Self: DefaultLogger {
     var value: Success {
         get async throws {
             do {
+                log.info("Starting execution for \(eventName)")
                 let valueReturned = try await execute()
+                log.info("Successfully completed execution for \(eventName) with result:\n\(prettify(valueReturned))")
```

Comment:
Do all Auth tasks execute this code path. If so, could `valueReturned` here (or `error` in the catch block below) contain any information that we would want to redact before logging?